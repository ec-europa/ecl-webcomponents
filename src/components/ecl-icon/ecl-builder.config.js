const path = require('path');
const pkg = require('./package.json');

const nodeModules = path.resolve(__dirname, '../../../node_modules');
const outputFolder = './build';

// SCSS includePaths
const includePaths = [nodeModules];

const banner = `${pkg.name} - ${pkg.version} Built on ${new Date().toISOString()}`;

module.exports = {
  copy: [
    {
      from: [path.resolve(nodeModules, '@ecl/resources-social-media-icons/dist/sprites/icons-social-media.svg')],
      to: path.resolve(outputFolder, 'images/ec'),
      options: { up: true },
    },
    {
      from: [path.resolve(nodeModules, '@ecl/resources-social-media-icons/dist/sprites/icons-social-media.svg')],
      to: path.resolve(outputFolder, 'images/eu'),
      options: { up: true },
    },
    {
      from: [path.resolve(nodeModules, '@ecl/resources-ec-icons/dist/sprites/icons.svg')],
      to: path.resolve(outputFolder, 'images/ec'),
      options: { up: true },
    },
    {
      from: [path.resolve(nodeModules, '@ecl/resources-eu-icons/dist/sprites/icon.svg')],
      to: path.resolve(outputFolder, 'images/eu'),
      options: { up: true },
    },
  ],
  styles: [
    {
      entry: path.resolve(__dirname, 'src/styles/ecl-icon-ec.scss'),
      dest: path.resolve(outputFolder, 'styles/ecl-icon-ec.css'),
      options: {
        banner,
        includePaths,
        sourceMap: false,
        minify: true,
      },
    },
    {
      entry: path.resolve(__dirname, 'src/styles/ecl-icon-eu.scss'),
      dest: path.resolve(outputFolder, 'styles/ecl-icon-eu.css'),
      options: {
        banner,
        includePaths,
        sourceMap: false,
        minify: true,
      },
    },
  ],
};
