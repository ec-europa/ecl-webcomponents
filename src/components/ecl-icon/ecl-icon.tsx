import { Component, Prop, h } from '@stencil/core';
import getAssetPath from "../../utils/assetPath";

@Component({
  tag: 'ecl-icon',
  styleUrls: {
    ec: './build/styles/ecl-icon-ec.css',
    eu: './build/styles/ecl-icon-eu.css',
  },
  shadow: false,
  scoped: true,
  assetsDirs: ['build'],
})

export class EclIcon {
  @Prop() styleClass: string = '';
  @Prop() icon: string = '';
  @Prop() size: string = 'xs';
  @Prop() color: string = '';
  @Prop() titleTag: string = '';
  @Prop({ mutable: true }) path: string;
  @Prop() transform: string = '';
  @Prop({ mutable: true }) theme: string;
  @Prop() sprite: string = '';

  componentWillLoad() {
    this.theme = document.documentElement.getAttribute('data-ecl-theme') ?? (this.theme || 'ec');
    
    if (!this.sprite) {
      this.path = getAssetPath(`./build/images/${this.theme}/icons.svg`);
    } else {
      this.path = getAssetPath(`./build/images/${this.theme}/${this.sprite}.svg`);
    }
  }

  getClass(): string {
    const styleClasses = [
      `ecl-icon`,
      `ecl-icon--${this.size}`,
      this.styleClass,
    ];
    if (this.color) {
      styleClasses.push(
        `ecl-icon--${this.color}`
      );
    }
    if (this.transform) {
      styleClasses.push(
        `ecl-icon--${this.transform}`
      );
    }

    return styleClasses.join(' ');
  }

  render() {
    return (
      <svg class={this.getClass()}>
      { this.titleTag ? <title>{this.titleTag}</title> : '' } 
        <use xlinkHref={`${this.path}#${this.icon}`}></use>
      </svg>
    )
  }
}