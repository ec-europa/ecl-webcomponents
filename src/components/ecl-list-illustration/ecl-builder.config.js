const path = require('path');
const pkg = require('./package.json');

const nodeModules = path.resolve(__dirname, './node_modules');
const outputFolder = './build';

// SCSS includePaths
const includePaths = [nodeModules];

const banner = `${pkg.name} - ${pkg.version} Built on ${new Date().toISOString()}`;

module.exports = {
  styles: [
    {
      entry: path.resolve(__dirname, 'src/styles/ecl-list-illustration-ec.scss'),
      dest: path.resolve(outputFolder, 'styles/ecl-list-illustration-ec.css'),
      options: {
        banner,
        includePaths,
        sourceMap: false,
        minify: true,
      },
    },
    {
      entry: path.resolve(__dirname, 'src/styles/ecl-list-illustration-eu.scss'),
      dest: path.resolve(outputFolder, 'styles/ecl-list-illustration-eu.css'),
      options: {
        banner,
        includePaths,
        sourceMap: false,
        minify: true,
      },
    },
  ],
};
