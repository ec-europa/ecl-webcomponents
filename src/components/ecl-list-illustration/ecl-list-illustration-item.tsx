import { Component, h, Prop} from '@stencil/core';

@Component({
  tag: 'ecl-list-illustration-item',
  shadow: false,
  scoped: false,
})

export class EclListIllustrationItem {
  @Prop() theme: string = 'ec';
  @Prop() itemTitle: string;
  @Prop() itemLink: string;
  @Prop() image: string;
  @Prop() icon: string;
  @Prop() imageAlt: string;
  @Prop() squareImage: boolean = false;
  @Prop() styleClass: string;
  @Prop() mediaSize: string = 'm';
  @Prop() itemValue: string;

  getClass(): string {
    const styleClasses = [
      `sc-ecl-list-illustration-${this.theme}`,
      `ecl-list-illustration__item`,
      this.styleClass
    ];

    return styleClasses.join(' ');
  }

  getImgClass(): string {
    const imgClasses = [
      `sc-ecl-list-illustration-${this.theme}`,
      `ecl-list-illustration__image`,
    ];

    if (this.squareImage) {
      imgClasses.push('ecl-list-illustration__image--square');
      imgClasses.push(`ecl-list-illustration__image--${this.mediaSize}`);
    }

    return imgClasses.join(' ');
  }

  getImgAttr() {
    const attrs = {
      role: 'img',
    }

    if (this.imageAlt) {
      attrs['aria-label'] = this.imageAlt;
    }

    return attrs;
  }

  getIconSize() {
    return this.mediaSize === 'l' ? '2xl' : 'l';
  }

  render() {
    return (
      <div class={this.getClass()}>
      { this.image ?
        <ecl-picture
          styleClass={`ecl-list-illustration__picture sc-ecl-list-illustration-${this.theme}`}
          imgClass={this.getImgClass()}
          image={this.image}
          imageAlt={this.imageAlt}
          {...this.getImgAttr()}
        >
          <slot name="sources"></slot>
        </ecl-picture> : ''
      }
        <div class={`ecl-list-illustration__detail sc-ecl-list-illustration-${this.theme}`}>
          <div class={`ecl-list-illustration__title-container sc-ecl-list-illustration-${this.theme}`}>
          { this.icon ?
            <ecl-icon
              icon={this.icon}
              size={this.getIconSize()}
              style-class={`ecl-list-illustration__icon sc-ecl-list-illustration-${this.theme}`}
            ></ecl-icon> : '' 
          }
          { this.itemValue ?
            <div class={`ecl-list-illustration__value sc-ecl-list-illustration-${this.theme}`}>{this.itemValue}</div> : ''
          }
          { this.itemTitle && (
            <div class={`ecl-list-illustration__title sc-ecl-list-illustration-${this.theme}`}>
              {this.itemLink ? (
                <ecl-link variant="standalone" path={this.itemLink}>{this.itemTitle}</ecl-link>
              ) : (
                this.itemTitle
              )}
            </div>
          )}
          </div>
          <div class={`ecl-list-illustration__description sc-ecl-list-illustration-${this.theme}`}>
            <slot></slot>
          </div>
        </div> 
      </div>
    );
  }
}
