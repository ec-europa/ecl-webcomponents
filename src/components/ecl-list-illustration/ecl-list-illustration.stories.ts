const getArgs = () => ({
  column: 1,
  zebra: false,
  squareImage: false,
  mediaSize: 'm',
  centered: false,
});

const getArgTypes = () => ({
  column: {
    control: { type: 'range', min: 1, max: 4, step: 1 },
    description: 'The number of column for the list (between 1 and 4)',
  },
  zebra: {
    control: { type: 'boolean' },
    description: 'Zebra styling',
  },
  squareImage: {
    name: 'squared images',
    control: { type: 'boolean' },
    description: 'Squared images',
  },
  mediaSize: {
    name: 'size of the media (for squared images only)',
    control: { type: 'select' },
    options: ['s', 'm', 'l'],
    description: 'Size of the squared images',
  },
  centered: {
    name: 'centered',
    control: { type: 'boolean' },
    description: 'Centered content in multiple columns layout',
  },
});

export default {
  title: 'Components/list with illustration',
};

const Template = (args) => 
  `<ecl-list-illustration
    theme="${args.theme}"
    zebra=${args.zebra}
    column="${args.column}"
    centered=${args.centered}
  >
    <ecl-list-illustration-item
      theme="${args.theme}"
      image="https://inno-ecl.s3.amazonaws.com/media/examples/example-image.jpg"
      item-title="List with illustration item 1"
      media-size="${args.mediaSize}"
      square-image=${args.squareImage}
      item-value="3.2 million"
    >
      Lorem ipsum dolor sit amet, consectetur adipiscing elit. Donec eleifend quam leo, at malesuada ex viverra vitae.
      Nullam id felis eu lorem condimentum rutrum vitae ut felis. Nam ultricies, metus vel aliquam euismod, lacus dolor sodales neque,
      in laoreet tellus erat posuere purus. Fusce sit amet sem dui. In nec lacinia eros.
    </ecl-list-illustration-item>
    <ecl-list-illustration-item
      theme="${args.theme}"
      image="https://inno-ecl.s3.amazonaws.com/media/examples/example-image2.jpg"
      item-title="List with illustration item 2"
      media-size="${args.mediaSize}"
      square-image=${args.squareImage}
      item-value="3.2 million"
    >
      Lorem ipsum dolor sit amet, consectetur adipiscing elit. Donec eleifend quam leo, at malesuada ex viverra vitae.
      Nullam id felis eu lorem condimentum rutrum vitae ut felis. Nam ultricies, metus vel aliquam euismod, lacus dolor sodales neque,
      in laoreet tellus erat posuere purus. Fusce sit amet sem dui. In nec lacinia eros.
    </ecl-list-illustration-item>
    <ecl-list-illustration-item
      theme="${args.theme}"
      image="https://inno-ecl.s3.amazonaws.com/media/examples/example-image3.jpg"
      item-title="List with illustration item 3"
      media-size="${args.mediaSize}"
      square-image=${args.squareImage}
      item-value="3.2 million"
    >
      Lorem ipsum dolor sit amet, consectetur adipiscing elit. Donec eleifend quam leo, at malesuada ex viverra vitae.
      Nullam id felis eu lorem condimentum rutrum vitae ut felis. Nam ultricies, metus vel aliquam euismod, lacus dolor sodales neque,
      in laoreet tellus erat posuere purus. Fusce sit amet sem dui. In nec lacinia eros.
    </ecl-list-illustration-item>
    <ecl-list-illustration-item
      theme="${args.theme}"
      image="https://inno-ecl.s3.amazonaws.com/media/examples/example-image4.jpg"
      item-title="List with illustration item 4"
      media-size="${args.mediaSize}"
      square-image=${args.squareImage}
      item-value="3.2 million"
    >
      Lorem ipsum dolor sit amet, consectetur adipiscing elit. Donec eleifend quam leo, at malesuada ex viverra vitae.
      Nullam id felis eu lorem condimentum rutrum vitae ut felis. Nam ultricies, metus vel aliquam euismod, lacus dolor sodales neque,
      in laoreet tellus erat posuere purus. Fusce sit amet sem dui. In nec lacinia eros.
    </ecl-list-illustration-item>
    <ecl-list-illustration-item
      theme="${args.theme}"
      image="https://inno-ecl.s3.amazonaws.com/media/examples/example-image5.jpg"
      item-title="List with illustration item 5"
      item-link="/example.html"
      media-size="${args.mediaSize}"
      square-image=${args.squareImage}
      item-value="3.2 million"
    >
      Lorem ipsum dolor sit amet, consectetur adipiscing elit. Donec eleifend quam leo, at malesuada ex viverra vitae.
      Nullam id felis eu lorem condimentum rutrum vitae ut felis. Nam ultricies, metus vel aliquam euismod, lacus dolor sodales neque,
      in laoreet tellus erat posuere purus. Fusce sit amet sem dui. In nec lacinia eros.
    </ecl-list-illustration-item>
  </ecl-list-illustration>`;

export const ListIllustration = Template.bind({});
ListIllustration.storyName = 'default';
ListIllustration.args = getArgs();
ListIllustration.argTypes = getArgTypes();
