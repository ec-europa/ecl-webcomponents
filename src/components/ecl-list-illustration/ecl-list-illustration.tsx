import { Component, h, Prop, Element} from '@stencil/core';

@Component({
  tag: 'ecl-list-illustration',
  styleUrls: {
    ec: './build/styles/ecl-list-illustration-ec.css',
    eu: './build/styles/ecl-list-illustration-eu.css',
  },
  shadow: false,
  scoped: true,
})

export class EclListIllustration {
  @Element() el: HTMLElement;
  @Prop() theme: string = 'ec';
  @Prop() variant: string = 'image';
  @Prop() zebra: boolean = false;
  @Prop() column: number = 1;
  @Prop() styleClass: string;
  @Prop() centered: boolean = false;

  getClass(): string {
    const styleClasses = [
      `ecl-list-illustration`,
      `ecl-list-illustration--col-${this.column}`,
      this.styleClass
    ];

    if (this.zebra) {
      styleClasses.push('ecl-list-illustration--zebra');
    }

    if (this.centered) {
      styleClasses.push('ecl-list-illustration--centered');
    }

    return styleClasses.join(' ');
  }

  componentDidRender() {
    // Clean the html to make the zebra work,.
    const items = this.el.querySelectorAll('.ecl-list-illustration__item');
    if (items) {
      this.el.querySelector('.ecl-list-illustration').innerHTML = '';
      this.el.querySelector('.ecl-list-illustration').append(...items);
    }
  }

  render() {
    return (
      <div class={this.getClass()}>
        <slot></slot>
      </div>
    );
  }
}
