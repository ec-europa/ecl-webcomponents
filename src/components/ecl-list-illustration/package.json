{
  "name": "@ecl/web-components-list-illustration",
  "author": "European Commission",
  "license": "EUPL-1.2",
  "version": "0.4.0",
  "description": "ECL Lst with illustration",
  "main": "ecl-list-illustration.tsx",
  "private": true,
  "scripts": {
    "build:styles": "cross-env NODE_ENV=production ecl-builder styles",
    "build": "run-s clean build:*",
    "clean": "rimraf build",
    "prepublish": "npm run build"
  },
  "devDependencies": {
    "@ecl/builder": "3.7.1",
    "@ecl/vanilla-component-list-illustration": "3.8.2",
    "cross-env": "7.0.3"
  },
  "repository": {
    "type": "git",
    "url": "git+https://github.com/planctus/ecl-webcomponents.git"
  },
  "bugs": {
    "url": "https://github.com/planctus/ecl-webcomponents/issues"
  },
  "homepage": "https://github.com/planctus/ecl-webcomponents",
  "keywords": [
    "ecl",
    "europa-component-library",
    "design-system",
    "web-components"
  ]
}
