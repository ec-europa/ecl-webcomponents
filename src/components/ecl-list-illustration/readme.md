# ecl-list-illustration



<!-- Auto Generated Below -->


## Properties

| Property      | Attribute      | Description | Type      | Default     |
| ------------- | -------------- | ----------- | --------- | ----------- |
| `icon`        | `icon`         |             | `string`  | `undefined` |
| `image`       | `image`        |             | `string`  | `undefined` |
| `imageAlt`    | `image-alt`    |             | `string`  | `undefined` |
| `itemTitle`   | `item-title`   |             | `string`  | `undefined` |
| `squareImage` | `square-image` |             | `boolean` | `false`     |
| `styleClass`  | `style-class`  |             | `string`  | `undefined` |
| `theme`       | `theme`        |             | `string`  | `'ec'`      |


## Dependencies

### Depends on

- [ecl-icon](../ecl-icon)

### Graph
```mermaid
graph TD;
  ecl-list-illustration-item --> ecl-icon
  style ecl-list-illustration-item fill:#f9f,stroke:#333,stroke-width:4px
```

----------------------------------------------

*Built with [StencilJS](https://stenciljs.com/)*
