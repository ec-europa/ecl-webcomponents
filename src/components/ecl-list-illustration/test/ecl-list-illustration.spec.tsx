import { newSpecPage } from '@stencil/core/testing';
import { EclListIllustration } from '../ecl-list-illustration';
import { EclListIllustrationItem } from '../ecl-list-illustration-item';
import { EclIcon } from '../../ecl-icon/ecl-icon';

describe('ecl-list-illustration', () => {
  it('renders', async () => {
    const page = await newSpecPage({
      components: [EclListIllustration, EclListIllustrationItem, EclIcon],
      html: `<ecl-list-illustration
        theme="ec"
        zebra=true
        column="1"
      >
        <ecl-list-illustration-item
          theme="ec"
          image="https://inno-ecl.s3.amazonaws.com/media/example.htmls/example.html-image.jpg"
          item-title="List with illustration item 1"
        >
          Lorem ipsum dolor sit amet, consectetur adipiscing elit. Donec eleifend quam leo, at malesuada ex viverra vitae.
          Nullam id felis eu lorem condimentum rutrum vitae ut felis. Nam ultricies, metus vel aliquam euismod, lacus dolor sodales neque,
          in laoreet tellus erat posuere purus. Fusce sit amet sem dui. In nec lacinia eros.
        </ecl-list-illustration-item>
        <ecl-list-illustration-item
          theme="ec"
          image="https://inno-ecl.s3.amazonaws.com/media/example.htmls/example.html-image.jpg"
          item-title="List with illustration item 2"
        >
          Lorem ipsum dolor sit amet, consectetur adipiscing elit. Donec eleifend quam leo, at malesuada ex viverra vitae.
          Nullam id felis eu lorem condimentum rutrum vitae ut felis. Nam ultricies, metus vel aliquam euismod, lacus dolor sodales neque,
          in laoreet tellus erat posuere purus. Fusce sit amet sem dui. In nec lacinia eros.
        </ecl-list-illustration-item>
        <ecl-list-illustration-item
          theme="ec"
          image="https://inno-ecl.s3.amazonaws.com/media/example.htmls/example.html-image.jpg"
          item-title="List with illustration item 3"
        >
          Lorem ipsum dolor sit amet, consectetur adipiscing elit. Donec eleifend quam leo, at malesuada ex viverra vitae.
          Nullam id felis eu lorem condimentum rutrum vitae ut felis. Nam ultricies, metus vel aliquam euismod, lacus dolor sodales neque,
          in laoreet tellus erat posuere purus. Fusce sit amet sem dui. In nec lacinia eros.
        </ecl-list-illustration-item>
        <ecl-list-illustration-item
          theme="ec"
          image="https://inno-ecl.s3.amazonaws.com/media/example.htmls/example.html-image.jpg"
          item-title="List with illustration item 4"
        >
          Lorem ipsum dolor sit amet, consectetur adipiscing elit. Donec eleifend quam leo, at malesuada ex viverra vitae.
          Nullam id felis eu lorem condimentum rutrum vitae ut felis. Nam ultricies, metus vel aliquam euismod, lacus dolor sodales neque,
          in laoreet tellus erat posuere purus. Fusce sit amet sem dui. In nec lacinia eros.
        </ecl-list-illustration-item>
        <ecl-list-illustration-item
          theme="ec"
          image="https://inno-ecl.s3.amazonaws.com/media/example.htmls/example.html-image.jpg"
          item-title="List with illustration item 5"
        >
          Lorem ipsum dolor sit amet, consectetur adipiscing elit. Donec eleifend quam leo, at malesuada ex viverra vitae.
          Nullam id felis eu lorem condimentum rutrum vitae ut felis. Nam ultricies, metus vel aliquam euismod, lacus dolor sodales neque,
          in laoreet tellus erat posuere purus. Fusce sit amet sem dui. In nec lacinia eros.
        </ecl-list-illustration-item>
      </ecl-list-illustration>`,
    });

    expect(page.root).toMatchSnapshot();
  });
});
