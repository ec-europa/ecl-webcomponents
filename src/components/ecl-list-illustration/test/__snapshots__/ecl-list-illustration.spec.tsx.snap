// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ecl-list-illustration renders 1`] = `
<ecl-list-illustration column="1" theme="ec" zebra="true">
  <!---->
  <div class="ecl-list-illustration ecl-list-illustration--col-1 ecl-list-illustration--zebra">
    <div class="ecl-list-illustration__item sc-ecl-list-illustration-ec">
      <div class="ecl-list-illustration__image sc-ecl-list-illustration-ec" role="img" style="background-image: url(https://inno-ecl.s3.amazonaws.com/media/examples/example-image.jpg);"></div>
      <div class="ecl-list-illustration__detail sc-ecl-list-illustration-ec">
        <div class="ecl-list-illustration__title-container sc-ecl-list-illustration-ec">
          <div class="ecl-list-illustration__title sc-ecl-list-illustration-ec">
            List with illustration item 1
          </div>
        </div>
        <div class="ecl-list-illustration__description sc-ecl-list-illustration-ec">
          Lorem ipsum dolor sit amet, consectetur adipiscing elit. Donec eleifend quam leo, at malesuada ex viverra vitae. Nullam id felis eu lorem condimentum rutrum vitae ut felis. Nam ultricies, metus vel aliquam euismod, lacus dolor sodales neque, in laoreet tellus erat posuere purus. Fusce sit amet sem dui. In nec lacinia eros.
        </div>
      </div>
    </div>
    <div class="ecl-list-illustration__item sc-ecl-list-illustration-ec">
      <div class="ecl-list-illustration__image sc-ecl-list-illustration-ec" role="img" style="background-image: url(https://inno-ecl.s3.amazonaws.com/media/examples/example-image.jpg);"></div>
      <div class="ecl-list-illustration__detail sc-ecl-list-illustration-ec">
        <div class="ecl-list-illustration__title-container sc-ecl-list-illustration-ec">
          <div class="ecl-list-illustration__title sc-ecl-list-illustration-ec">
            List with illustration item 2
          </div>
        </div>
        <div class="ecl-list-illustration__description sc-ecl-list-illustration-ec">
          Lorem ipsum dolor sit amet, consectetur adipiscing elit. Donec eleifend quam leo, at malesuada ex viverra vitae. Nullam id felis eu lorem condimentum rutrum vitae ut felis. Nam ultricies, metus vel aliquam euismod, lacus dolor sodales neque, in laoreet tellus erat posuere purus. Fusce sit amet sem dui. In nec lacinia eros.
        </div>
      </div>
    </div>
    <div class="ecl-list-illustration__item sc-ecl-list-illustration-ec">
      <div class="ecl-list-illustration__image sc-ecl-list-illustration-ec" role="img" style="background-image: url(https://inno-ecl.s3.amazonaws.com/media/examples/example-image.jpg);"></div>
      <div class="ecl-list-illustration__detail sc-ecl-list-illustration-ec">
        <div class="ecl-list-illustration__title-container sc-ecl-list-illustration-ec">
          <div class="ecl-list-illustration__title sc-ecl-list-illustration-ec">
            List with illustration item 3
          </div>
        </div>
        <div class="ecl-list-illustration__description sc-ecl-list-illustration-ec">
          Lorem ipsum dolor sit amet, consectetur adipiscing elit. Donec eleifend quam leo, at malesuada ex viverra vitae. Nullam id felis eu lorem condimentum rutrum vitae ut felis. Nam ultricies, metus vel aliquam euismod, lacus dolor sodales neque, in laoreet tellus erat posuere purus. Fusce sit amet sem dui. In nec lacinia eros.
        </div>
      </div>
    </div>
    <div class="ecl-list-illustration__item sc-ecl-list-illustration-ec">
      <div class="ecl-list-illustration__image sc-ecl-list-illustration-ec" role="img" style="background-image: url(https://inno-ecl.s3.amazonaws.com/media/examples/example-image.jpg);"></div>
      <div class="ecl-list-illustration__detail sc-ecl-list-illustration-ec">
        <div class="ecl-list-illustration__title-container sc-ecl-list-illustration-ec">
          <div class="ecl-list-illustration__title sc-ecl-list-illustration-ec">
            List with illustration item 4
          </div>
        </div>
        <div class="ecl-list-illustration__description sc-ecl-list-illustration-ec">
          Lorem ipsum dolor sit amet, consectetur adipiscing elit. Donec eleifend quam leo, at malesuada ex viverra vitae. Nullam id felis eu lorem condimentum rutrum vitae ut felis. Nam ultricies, metus vel aliquam euismod, lacus dolor sodales neque, in laoreet tellus erat posuere purus. Fusce sit amet sem dui. In nec lacinia eros.
        </div>
      </div>
    </div>
    <div class="ecl-list-illustration__item sc-ecl-list-illustration-ec">
      <div class="ecl-list-illustration__image sc-ecl-list-illustration-ec" role="img" style="background-image: url(https://inno-ecl.s3.amazonaws.com/media/examples/example-image.jpg);"></div>
      <div class="ecl-list-illustration__detail sc-ecl-list-illustration-ec">
        <div class="ecl-list-illustration__title-container sc-ecl-list-illustration-ec">
          <div class="ecl-list-illustration__title sc-ecl-list-illustration-ec">
            List with illustration item 5
          </div>
        </div>
        <div class="ecl-list-illustration__description sc-ecl-list-illustration-ec">
          Lorem ipsum dolor sit amet, consectetur adipiscing elit. Donec eleifend quam leo, at malesuada ex viverra vitae. Nullam id felis eu lorem condimentum rutrum vitae ut felis. Nam ultricies, metus vel aliquam euismod, lacus dolor sodales neque, in laoreet tellus erat posuere purus. Fusce sit amet sem dui. In nec lacinia eros.
        </div>
      </div>
    </div>
  </div>
</ecl-list-illustration>
`;
